library/cipher.c

```diff
@@ -215,10 +215,15 @@ int mbedtls_cipher_set_iv( mbedtls_cipher_context_t *ctx,
                    const unsigned char *iv, size_t iv_len )
 {
     size_t actual_iv_size;
-
-    if( NULL == ctx || NULL == ctx->cipher_info || NULL == iv )
+    if( NULL == ctx || NULL == ctx->cipher_info ||
+        ( NULL == iv && ( ctx->cipher_info->mode != MBEDTLS_MODE_ECB ) ) )
```

Comment:
I would suggest splitting the condition to make it more manageable. For example:
```
if( NULL == ctx || NULL == ctx->cipher_info )
    return( MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA );
else if( NULL == iv && ctx->cipher_info->mode != MBEDTLS_MODE_ECB )
    return( MBEDTLS_ERR_CIPHER_BAD_INPUT_DATA );
```

Reply:
The documentation says that when the IV is not needed then `iv == NULL and iv_len == 0`. Should we ensure that both parameters are correct?

Reply:
ok